core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/ArtifactSignerFactory.java

```diff
@@ -135,6 +142,30 @@ private ArtifactSigner createHashicorpArtifact(final HashicorpSigningMetadata me
     }
   }
 
+  private ArtifactSigner createAzureArtifact(final AzureSigningMetadata metadata) {
+
+    try {
+      final AzureKeyVault azureVault =
+          new AzureKeyVault(
+              metadata.getClientId(),
+              metadata.getClientSecret(),
+              metadata.getTenantId(),
+              metadata.getVaultName());
+
+      final Optional<String> secret = azureVault.fetchSecret(metadata.getSecretName());
+      if (secret.isEmpty()) {
+        throw new SigningMetadataException(
+            "secret '" + metadata.getSecretName() + "' doesn't exist");
+      }
+      final BLSKeyPair keyPair =
```

Comment:
This is part of overall "caching". It will ultimately be called from 

tech/pegasys/eth2signer/core/Runner.java:133

~~~
final DirectoryBackedArtifactSignerProvider blsSignerProvider =
          createSignerProvider(metricsSystem, vertx);
      blsSignerProvider.cacheAllSigners();
~~~

At the moment, it is only called once at the startup.

Reply:
It is expected that the factory creates the signer each time as the caching is done at the DirectoryBackedArtifactSignerProvider layer. If caching is enabled the DirectoryBackedArtifactSignerProvider will ensure that this only gets created once. It is possible to disable caching in which case this will retrieve the key each time.